vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleInformationMain.java

```diff
@@ -0,0 +1,131 @@
+ * {@link Experiment} of a {@link life.qbic.projectmanagement.domain.project.Project} information
+ * via the provided {@link life.qbic.projectmanagement.domain.project.ProjectId} in the URL
+ */
+
+@Route(value = "projects/:projectId?/samples", layout = ProjectViewPage.class)
+@SpringComponent
+@UIScope
+@PermitAll
+public class SampleInformationMain extends MainComponent implements BeforeEnterObserver,
+    RouterLayout {
+
+  @Serial
+  private static final long serialVersionUID = 3778218989387044758L;
+  private static final Logger log = LoggerFactory.logger(SampleInformationMain.class);
+  private final ProjectNavigationBarComponent projectNavigationBarComponent;
+  private final transient SampleInformationMainHandler sampleInformationMainHandler;
+  public static final String PROJECT_ID_ROUTE_PARAMETER = "projectId";
+
+  public SampleInformationMain(
+      @Autowired ProjectNavigationBarComponent projectNavigationBarComponent,
+      @Autowired SampleContentComponent sampleContentComponent,
+      @Autowired SampleSupportComponent sampleSupportComponent) {
+    super(sampleContentComponent, sampleSupportComponent);
+    Objects.requireNonNull(projectNavigationBarComponent);
+    Objects.requireNonNull(sampleContentComponent);
+    Objects.requireNonNull(sampleSupportComponent);
+    this.projectNavigationBarComponent = projectNavigationBarComponent;
+    layoutComponent();
+    sampleInformationMainHandler = new SampleInformationMainHandler(projectNavigationBarComponent,
+        sampleContentComponent, sampleSupportComponent);
+    log.debug(String.format(
+        "\"New instance for Sample Information page (#%s) created with Project Navigation Bar Component (#%s), Sample Content Component (#%s) and Sample Support Component (#%s)",
+        System.identityHashCode(this), System.identityHashCode(projectNavigationBarComponent),
+        System.identityHashCode(sampleContentComponent),
+        System.identityHashCode(sampleSupportComponent)));
+  }
+
+  private void layoutComponent() {
+    addClassName("sample");
+    addComponentAsFirst(projectNavigationBarComponent);
+  }
+
+  /**
+   * Provides the {@link ProjectId} to the components within this page
+   * <p>
+   * This method serves as an entry point providing the necessary {@link ProjectId} to the
+   * components within this cage
+   *
+   * @param projectId projectId of the selected project
+   */
+  public void projectId(ProjectId projectId) {
+    sampleInformationMainHandler.setProjectId(projectId);
+  }
+
+  /**
+   * Callback executed before navigation to attaching Component chain is made.
+   *
+   * @param event before navigation event with event details
+   */
+  @Override
+  public void beforeEnter(BeforeEnterEvent event) {
+    event.getRouteParameters().get(PROJECT_ID_ROUTE_PARAMETER)
+        .ifPresent(sampleInformationMainHandler::propagateProjectId);
+  }
+
+  private final class SampleInformationMainHandler {
```

Comment:
I think we decided against having a separate handler class inside the component. In this case this adds unnecessary complexity as you call 
```text
projectId(id) -> hanlder.setProjectId(id) -> childComponent.setProjectId(id) AND beforeEnter -> handler.propagateId(id) -> component.handler.propagateId(id) -> child components.setid(id)
```

Reply:
Oh then we need to do this for all our classes. I'll adapt it! Thanks for the headsup 😅